modules/wpcom-block-editor/class-jetpack-wpcom-block-editor.php

```diff
@@ -55,6 +56,22 @@ public function is_iframed_block_editor() {
 		return ( 'post.php' === $pagenow || 'post-new.php' === $pagenow ) && ! empty( $_GET['frame-nonce'] ) && $is_calypsoify;
 	}
 
+	/**
+	 * Shows a custom message if the user is logged out.
+	 *
+	 * The iframed block editor can be only embedded in WordPress.com if the user is logged
+	 * into the Jetpack site. So we abort the default redirection to the login page (which
+	 * cannot be embedded in a iframe) and instead we explain that we need the user to log
+	 * into Jetpack.
+	 */
+	public function show_error_if_logged_out() {
+		if ( ! get_current_user_id() ) {
+			$login_url = wp_login_url();
+			/* translators: %s: Login URL */
+			wp_die( sprintf( __( 'In order to use the block editor of this Jetpack site in WordPress.com, you need to <a href="%s" target="_blank" rel="noopener noreferrer">log into</a> the Jetpack site.', 'jetpack' ), $login_url ) ); // // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
I think by default `wp_die` returns HTTP status code [`500 Internal Server Error`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500) — tho I didn't check if this is still the case.

I don't know if that can affect how we collect stats in different layers.

Something like [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) or [403](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) would prolly be better here.

https://developer.wordpress.org/reference/functions/wp_die/

Reply:
Good catch, I'll spin up a PR for it

Reply:
See https://github.com/Automattic/jetpack/pull/12162